### PR TITLE
[DOCS-4912] Update log docs to use SSL/TLS instead of telnet

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do?

- Updates logs docs to use SSL/TLS instead of telnet

### Motivation

[DOCS-4912](https://datadoghq.atlassian.net/browse/DOCS-4912), PM/Eng request.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-4912]: https://datadoghq.atlassian.net/browse/DOCS-4912?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/logs/log_collection/_index.md [hunk 1]
```diff
@@ -218,37 +218,56 @@ You can send logs to Datadog platform over HTTP. Refer to the [Datadog Log HTTP
...
-<DATADOG_API_KEY> Log sent directly via TCP
+```shell
+gnutls-cli intake.logs.datadoghq.com 10514
 ```
 
-Your payload, or `Log sent directly via TCP` as written in the example, can be in raw, Syslog, or JSON format. If your payload is in JSON format, Datadog automatically parses its attributes.
+For OpenSSL, run the following command:
+
+```shell
+openssl s_client -connect intake.logs.datadoghq.com 10514
```
Comment (trazfr) [inline content/en/logs/log_collection/_index.md:230]:
```suggestion
openssl s_client -connect intake.logs.datadoghq.com:10516
```

File: content/en/logs/log_collection/_index.md [hunk 2]
```diff
@@ -218,37 +218,56 @@ You can send logs to Datadog platform over HTTP. Refer to the [Datadog Log HTTP
...
 {{< site-region region="us" >}}
 
-Test it manually with telnet. You must prefix the log entry with your [Datadog API Key][1] and add a payload.
+You can manually test your connection using OpenSSL, GnuTLS, or another SSL/TLS client. For GnuTLS, run the following command:
 
-```text
-telnet intake.logs.datadoghq.com 10514
-<DATADOG_API_KEY> Log sent directly via TCP
+```shell
+gnutls-cli intake.logs.datadoghq.com 10514
```
Comment (trazfr) [inline content/en/logs/log_collection/_index.md:224]:
```suggestion
gnutls-cli intake.logs.datadoghq.com:10516
```

File: content/en/logs/guide/log-collection-troubleshooting-guide.md [hunk 3]
```diff
@@ -26,12 +26,19 @@ Changes in the configuration of the `datadog-agent` won't be taken into account
...
-* `openssl s_client -connect intake.logs.datadoghq.com:10516`
-* `telnet intake.logs.datadoghq.com 10514`
+```shell
+openssl s_client -connect intake.logs.datadoghq.com:10516
+```
+
+For GnuTLS, run the following command:
+
+```shell
+gnutls-cli intake.logs.datadoghq.com 10516
```
Comment (trazfr) [inline content/en/logs/guide/log-collection-troubleshooting-guide.md:40]:
```suggestion
gnutls-cli intake.logs.datadoghq.com:10516
```

File: content/en/logs/log_collection/_index.md [hunk 4]
```diff
@@ -218,37 +218,56 @@ You can send logs to Datadog platform over HTTP. Refer to the [Datadog Log HTTP
...
 {{< site-region region="eu" >}}
 
-Test it manually with telnet. You must prefix the log entry with your [Datadog API Key][1] and add a payload.
+You can manually test your connection using OpenSSL, GnuTLS, or another SSL/TLS client. For GnuTLS, run the following command:
 
-```text
-telnet agent-intake.logs.datadoghq.eu 443
-<DATADOG_API_KEY> Log sent directly via TCP
+```shell
+gnutls-cli tcp-intake.logs.datadoghq.eu 443
```
Comment (trazfr) [inline content/en/logs/log_collection/_index.md:253]:
```suggestion
gnutls-cli tcp-intake.logs.datadoghq.eu:443
```

File: content/en/logs/log_collection/_index.md [hunk 5]
```diff
@@ -218,37 +218,56 @@ You can send logs to Datadog platform over HTTP. Refer to the [Datadog Log HTTP
...
+```
+
+For OpenSSL, run the following command:
+
+```shell
+openssl s_client -connect tcp-intake.logs.datadoghq.eu 443
```
Comment (trazfr) [inline content/en/logs/log_collection/_index.md:259]:
```suggestion
openssl s_client -connect tcp-intake.logs.datadoghq.eu:443
```

File: content/en/logs/guide/docker-logs-collection-troubleshooting-guide.md [hunk 6]
```diff
@@ -137,12 +137,19 @@ If the Logs Agent status looks like the example in [Check the Agent status](#che
...
-* `openssl s_client -connect intake.logs.datadoghq.com:10516`
-* `telnet intake.logs.datadoghq.com 10514`
+```shell
+openssl s_client -connect intake.logs.datadoghq.com:10516
+```
+
+For GnuTLS, run the following command:
+
+```shell
+gnutls-cli intake.logs.datadoghq.com 10516
```
Comment (trazfr) [inline content/en/logs/guide/docker-logs-collection-troubleshooting-guide.md:151]:
```suggestion
gnutls-cli intake.logs.datadoghq.com:10516
```

File: content/en/logs/guide/log-collection-troubleshooting-guide.md [hunk 7]
```diff
@@ -26,20 +26,27 @@ Changes in the configuration of the `datadog-agent` won't be taken into account
...
+gnutls-cli intake.logs.datadoghq.com:10516
+```
 
 And then by sending a log like the following:
 
 ```text
 <API_KEY> this is a test message
 ```
 
-- If opening the port 10514 or 10516 is not an option, it is possible to configure the Datadog Agent to send logs through HTTPS by adding the following in `datadog.yaml`:
```
Comment (maycmlee) [inline content/en/logs/guide/log-collection-troubleshooting-guide.md:42]: @trazfr could you just confirm that it makes sense to remove this from the doc as well?

Comment (trazfr) [reply content/en/logs/guide/log-collection-troubleshooting-guide.md:42]: As you wish.

It makes sense as we no longer ask new customers to use it. There are only legacy implementations which use 10514.
On the other hand, we could keep it customers using the legacy non-TLS port, even if it is discouraged.

**Review by trazfr — APPROVED** [review]

File: content/en/logs/guide/docker-logs-collection-troubleshooting-guide.md [hunk 8]
```diff
@@ -137,20 +137,27 @@ If the Logs Agent status looks like the example in [Check the Agent status](#che
...
 
-The Datadog Agent sends its logs to Datadog over tcp via port 10516. If that connection is not available, logs fail to be sent and an error is recorded in the `agent.log` file to that effect.
+The Datadog Agent sends its logs to Datadog over TCP using port 10516. If that connection is not available, logs fail to be sent and an error is recorded in the `agent.log` file to that effect.
 
-Test manually your connection by running a telnet or openssl command like so (port 10514 would work too, but is less secure):
+You can manually test your connection using OpenSSL, GnuTLS, or another SSL/TLS client. For OpenSSL, run the following command:
 
-* `openssl s_client -connect intake.logs.datadoghq.com:10516`
-* `telnet intake.logs.datadoghq.com 10514`
+```shell
```
Comment (urseberry) [inline content/en/logs/guide/docker-logs-collection-troubleshooting-guide.md:144]: Hooray for using the new shell functionality!

**Review by urseberry — APPROVED** [review]

**Pull request merged**